macros/Compat.ejs

```diff
@@ -1,66 +1,399 @@
 
 */
 
-var githubURL = "https://raw.githubusercontent.com/mdn/browser-compat-data/master/";
-var locale = env.locale;
-var output = "";
-var file = $0 + ".json";
-var leaf = $1;
-var url = githubURL + file;
+var bcd = require('mdn-browser-compat-data');
 
-var noDataFoundl10n = mdn.localString({
-    "en-US": 'No compatibility data found. Please contribute data for "' + env.title + '" to the <a href="https://github.com/mdn/browser-compat-data">MDN compatibility data repository</a>.',
-    "fr"   : "Aucune donnée de compatibilité trouvée"
+var query = $0;
+var output = '';
+
+var compatStrings = string.deserialize(template("L10n:CompatTable"));
+var localize = mdn.getLocalString;
+
+var s_no_data_found = mdn.localString({
+    'en-US': `No compatibility data found. Please contribute data for "${query}" to the <a href="https://github.com/mdn/browser-compat-data">MDN compatibility data repository</a>.`
 });
 
+const desktopBrowsers = {
+  chrome: "Chrome",
+  firefox: "Firefox",
+  edge: "Edge",
+  ie: "Internet Explorer",
+  opera: "Opera",
+  safari: "Safari",
+  servo: "Servo"
+};
+
+const mobileBrowsers = {
+  webview_android: "Android",
+  chrome_android: "Chrome for Android",
+  edge_mobile: "Edge mobile",
+  firefox_android: "Firefox for Android",
+  ie_mobile: "IE mobile",
+  opera_android: "Opera Android",
+  safari_ios: "iOS Safari",
+}
+
+var notesArray = [];
 
-if ($0) {
-  try {
-    var data = mdn.fetchJSONResource(url);
-  } catch (e) {
-    output = 'File "' + file + '" not found or malformed.'
+/*
+Write the table header.
+*/
+function writeTableHead(environment, browserNames) {
+  let browserNameKeys = Object.keys(browserNames);
+  let output = `<div id="compat-${environment}"><table class="compat-table"><thead><tr>`;
+  output +=  `<th>${localize(compatStrings, 'feature')}</th>`
+  for (let browserNameKey of browserNameKeys) {
+    output += `<th>${browserNames[browserNameKey]}</th>`;
   }
+  output += "</tr></thead>";
+  return output;
 }
 
-if (data) {
-    if (leaf !== '') {
-      output = template("CompatDataTable", [data[leaf], "api"]);
+/*
+Given the value of `version_added` or `version_removed`, this returns:
+  * a string to appear in the table cell, like "Yes", "No" or "?"
+
+*/
+function getVersionString(versionInfo) {
+  switch (versionInfo) {
+    case null:
+      return `<span title="${localize(compatStrings, 'supportsShort_unknown_title')}"
+              style="color: rgb(255, 153, 0);">${localize(compatStrings, 'supportsShort_unknown')}</span>`;
+      break;
+    case true:
+      return `<span style="color: #888">(${localize(compatStrings, 'supportsShort_yes')})</span>`;
+      break;
+    case false:
+      return `<span style="color: #f00">(${localize(compatStrings, 'supportsShort_no')})</span>`;
+      break;
+    default:
+      return versionInfo;
+  }
+}
+
+/*
+  Create the note for a browser flag or preference
+*/
+function generateFlagsNote(supportData, browserId) {
```

Comment:
I think it would be worth trying to refactor this function:

* it could make good use of template literals
* it uses single quotes, where most of the macro uses double quotes (I don't think it matter which we use, but we should be consistent)
* there's a fair amount of repetition (e.g. `((typeof(supportData.version_removed) === 'string')?'was':'is')` 4 times)
* some very long lines
* some quite complex conditionals

It might be worthwhile trying to reorganize some of the output so you're not crafting custom strings for every different option. For example, you could have some const text describing how to change prefs in Firefox or Chrome:

    const firefoxPrefs = "To change preferences on Firefox, visit about:config"
    const chromePrefs = "To change preferences on Chrome, visit chrome://flags"

... then craft a string common across all browsers like: "this feature is available behind the `EnableToadstools` preference".

But again, maybe this can wait until the table redesign.